src/core/execute.c

```diff
@@ -2735,6 +2752,28 @@ static void invalid_env(const char *p, void *userdata) {
         log_unit_error(info->unit, "Ignoring invalid environment assignment '%s': %s", p, info->path);
 }
 
+int exec_context_named_iofds(Unit *unit, const ExecContext *c, const ExecParameters *p, int *named_iofds) {
+        unsigned int i, targets = 0;
+        targets = (c->std_input == EXEC_INPUT_NAMED_FD) +
+                       (c->std_output == EXEC_OUTPUT_NAMED_FD) +
+                       (c->std_error == EXEC_OUTPUT_NAMED_FD);
+        for (i = 0; i < p->n_fds && targets > 0; i++) {
+                if (named_iofds[STDIN_FILENO] < 0 && c->std_input == EXEC_INPUT_NAMED_FD && streq(p->fd_names[i], "StandardInput")) {
+                        named_iofds[STDIN_FILENO] = p->fds[i];
+                        targets--;
+                }
+                else if (named_iofds[STDOUT_FILENO] < 0 && c->std_output == EXEC_OUTPUT_NAMED_FD && streq(p->fd_names[i], "StandardOutput")) {
+                        named_iofds[STDOUT_FILENO] = p->fds[i];
+                        targets--;
+                }
```

Comment:
minor coding style nitpick: please place `}` and the `else if…` on the same line